### PR TITLE
update TDD and doubles explanation

### PR DESCRIPTION
**Pull request by rlmoser99**
In reviewing file 13, I changed the documentation link to be the one that @aaron-contreras suggested, because it clearly talked about method stubs returning a value. In addition, I move the 'Testing For Beginners' link to this file, because it gives an extended explantation on stubs.

In reviewing file 14, I updated the explanation at the top regarding using doubles in TDD. I liked the explanation of examples in the 'RSpec mocks and stubs in plain English' article that @aaron-contreras suggested, so I added it as an additional resource.

In addition, in file 14, I expanded on the typical Red-Green-Refactor workflow. Since, the assignment already breaks down the workflow for each of the 3 methods to be write a test, write the method, then write the rest of the tests. Explicitly writing out the extended workflow sets the expectations of the TDD process.

### CODE REVIEW

File: spec/14_find_number_spec.rb [hunk 1]
```diff
@@ -4,29 +4,29 @@
...
-# Doubles are useful in TDD because you can create test functionality that is not coded yet.
+# Doubles are very useful in TDD because you can create test functionality that is not coded yet.
 
-# Doubles can be a confusing concept, so here are several resources to check out:
-# https://www.tutorialspoint.com/rspec/rspec_test_doubles.htm
-# https://relishapp.com/rspec/rspec-mocks/v/3-9/docs/basics/test-doubles
-# http://testing-for-beginners.rubymonstas.org/test_doubles.html
-
-# In this file, we will be testing the 'FindNumber' class.
+# In this example, we will testing the class 'FindNumber'. Look at the lib/14_find_number.rb file.
```
Comment (aaron-contreras) [inline spec/14_find_number_spec.rb:12]: Would read better as "we will **be** testing the ..."

Comment (rlmoser99) [reply spec/14_find_number_spec.rb:12]: Yes, I will make this change. Thank you for reading through these changes to catch anything like this.
